horovod/mxnet/mpi_ops.cc

```diff
@@ -290,8 +290,7 @@ extern "C" int horovod_mxnet_broadcast_async(NDArray* input, NDArray* output,
     DoBroadcast(input, output, root_rank, op_name, on_complete);
   };
 
-#if HAVE_CUDA && HOROVOD_GPU_BROADCAST != 'M'
-  ThrowIfError(common::CheckInitialized());
```

Comment:
We should keep this check here since it's not done in `DoBroadcastCudaOnCPU`.

Reply:
I have moved this into DoBroadcastCudaOnCPU to be consistent with other APIs.